packages/vite/src/node/preview.ts

```diff
@@ -53,17 +57,30 @@ export async function preview(
   const logger = config.logger
   const base = config.base
 
-  httpServer.listen(port, hostname.host, () => {
-    logger.info(
-      chalk.cyan(`\n  vite v${require('vite/package.json').version}`) +
-        chalk.green(` build preview server running at:\n`)
-    )
+  await httpServerStart(httpServer, {
+    port,
+    strict: !!serverOptions.port,
+    host: hostname.host,
+    logger
+  })
+    .then((serverPort) => {
+      logger.info(
+        chalk.cyan(`\n  vite v${require('vite/package.json').version}`) +
+          chalk.green(` build preview server running at:\n`)
+      )
```

Comment:
We don't need to use `.then` here, we can get the `serverPort value and use it. The catch at the end is also not needed in this case. Don't commit the suggestion, it is just to illustrate the comment (GitHub doesn't let me go over the deleted lines)
```suggestion
  const serverPort = await httpServerStart(httpServer, {
    port,
    strict: !!serverOptions.port,
    host: hostname.host,
    logger
  })
  
  logger.info(
    chalk.cyan(`\n  vite v${require('vite/package.json').version}`) +
      chalk.green(` build preview server running at:\n`)
  )
```